python-interface/test/examples/fashion-mnist/test_vector_fashion-mnist.py

```diff
@@ -0,0 +1,63 @@
+# ==============================================================================
+#  This file is part of the SPNC project under the Apache License v2.0 by the
+#  Embedded Systems and Applications Group, TU Darmstadt.
+#  For the full copyright and license information, please view the LICENSE
+#  file that was distributed with this source code.
+#  SPDX-License-Identifier: Apache-2.0
+# ==============================================================================
+
+import numpy as np
+import os
+import shutil
+import time
+from spnc.cpu import CPUCompiler
+from xspn.serialization.binary.BinarySerialization import BinaryDeserializer
+
+
+def test_vector_fashion_mnist():
+    if not CPUCompiler.isVectorizationSupported():
+        print("Test not supported by the compiler installation")
+        return 0
+    # Locate test resources located in same directory as this script.
+    scriptPath = os.path.realpath(os.path.dirname(__file__))
+
+    # Deserialize model
+    model = BinaryDeserializer(
+        os.path.join(scriptPath, "fashion_mnist_200_100_4_5_10_9_1_True.bin")).deserialize_from_file()
+    spn = model.root
+
+    inputs = np.genfromtxt(os.path.join(scriptPath, "input.csv"), delimiter=",", dtype="float64")
+    reference = np.genfromtxt(os.path.join(scriptPath, "fashion_mnist_200_100_4_5_10_9_1_True_output.csv"),
+                              delimiter=",",
+                              dtype="float64")
+    reference = reference.reshape(1000)
+    # Compile the kernel.
+    options = dict()
+    options["slp-max-look-ahead"] = 10
+    options["slp-max-node-size"] = 10000
+    options["slp-max-attempts"] = 5
+    options["slp-max-successful-iterations"] = 1
+    options["slp-reorder-dfs"] = True
+    options["slp-allow-duplicate-elements"] = False
+    options["slp-allow-topological-mixing"] = False
+    options["slp-use-xor-chains"] = True
+    compiler = CPUCompiler(vectorize=True, computeInLogSpace=True, vectorLibrary="LIBMVEC", **options)
+    kernel = compiler.compile_ll(spn=spn, batchSize=1, supportMarginal=False)
+    shutil.copyfile(kernel.fileName(), os.path.join(scriptPath, "fashion.so"))
```

Comment:
Why is the kernel copied to `fashion.so`, I did not find any other mention of that file in the script.

Reply:
Oh these copies took place because I wanted to analyze the object code during development without having to look for the kernels in the /tmp/ directory all the time. Fixed in 59632c8e28180712a8260a164213a1520b832d9a.